### ISSUE TITLE
Question regarding Upgrades

### ISSUE THREAD

**Issue by JamsRepos** [shemanaev/jellyfin-plugin-media-cleaner#47]
Just a question; what happens when sonarr/radarr upgrades a file due to having a better quality, score, etc... Does this then reset the duration of content being added?

**Comment by shemanaev** [comment]
If you talk about unplayed media, I didn't make an experiment on this
specifically, but you can do it yourself: watch for a field "Date added" in
the metadata editor.
Played/in-process doesn't care about file updates.

вс, 28 апр. 2024 г. в 03:54, Jam ***@***.***>:

**Comment by JamsRepos** [comment]
Yeah - the Date added resets in this case. Which means the plugin will be resetting upon an upgrade, which is a shame. 

Is there anything we can do to combat this or a Jellyfin setting at all we can turn on to stop things being re-added on an upgrade?

**Comment by shemanaev** [comment]
Does "Date added behavior for new content" option set correctly?
![изображение](https://github.com/shemanaev/jellyfin-plugin-media-cleaner/assets/1058537/34c1bda4-363c-4fc4-97d8-8d199c758b24)

**Comment by JamsRepos** [comment]
Mine is set to the same option as you have.

**Comment by shemanaev** [comment]
I don't see a workaround for this behavior then...

**Comment by Yankees4life** [comment]
> Yeah - the Date added resets in this case. Which means the plugin will be resetting upon an upgrade, which is a shame.
> 
> Is there anything we can do to combat this or a Jellyfin setting at all we can turn on to stop things being re-added on an upgrade?

If you're using notifarr, you can enable the setting under Jellyfin settings to unmonitor shows after being watched but that's about it. 

Also, you can use trakt to set any show as watched and then let it run the task to import watched status

**Comment by JamsRepos** [comment]
> > Yeah - the Date added resets in this case. Which means the plugin will be resetting upon an upgrade, which is a shame.
> > Is there anything we can do to combat this or a Jellyfin setting at all we can turn on to stop things being re-added on an upgrade?
> 
> If you're using notifarr, you can enable the setting under Jellyfin settings to unmonitor shows after being watched but that's about it.
> 
> Also, you can use trakt to set any show as watched and then let it run the task to import watched status

I think you're talking about something else. We are talking about when content is upgraded (i.e there's a better file found by sonarr/radarr) then when Jellyfin scans the new file into the library, it is counted as a new film. So, that would reset this plugins data as it wouldn't be checking anymore from when the film was first added but now when it was upgraded.

**Comment by NicolaSmaniotto** [comment]
Hello everyone. The reason for this issue is that when a file is modified and imported again in Jellyfin, it is seen as a new addition, regardless of previous metadata (in a way that data refers to the file, not the abstract "movie" item).

To fix this, I have written a (disgusting) python script that pulls the metadata from the API and saves the addition date in an external database. If the file is later re-added to Jellyfin, its addition date is overwritten with the older one. I've been using it for a few weeks and seems to work as it should. It's currently just a file that I start with the server (through systemd) and keeps running on its own. I'll look into packaging it in a better way, maybe with a simple container, but for the moment it's a sufficient workaround for my issue.

Here is a gist if you want to try it yourselves: https://gist.github.com/NicolaSmaniotto/aff7d01ea7055c111c85ce1828fc7ed4

**Comment by surepointit** [comment]
Where would we run this script from? I run Jellyfin in a docker container. Inside the container or on the server that's running docker?

**Comment by NicolaSmaniotto** [comment]
You can run it on the server or on an entirely different machine, just set the appropriate variables at lines 21-23.

It would be hard to run it inside the Jellyfin container, as the official image does not ship with python, and any modification would be overwritten by updates. I've created a basic systemd unit file that runs it at boot

**Comment by surepointit** [comment]
Ok, I created this python script on my Ubuntu vm with docker on it that has Jellyfin. I have python3 installed and ran the script and get the following messages. Something about a KeyError: 'ProductionYear' Screenshot attached. What am I doing wrong?

![Screenshot 2024-12-13 164114](https://github.com/user-attachments/assets/760bda1d-9ad1-4e8c-a33c-4f73d01e8cdd)

**Comment by NicolaSmaniotto** [comment]
Apparently you have some movie without a release year, but the script tries to read it. I haven't encountered missing metadata like that and since there is no check in place, the script just crashes. You can probably solve this by putting `year = item.get("ProductionYear", -1)` at line 43 (if a movie is missing the year, it just uses -1).

Side note, it's probably better to write these things under the gist or writing me an email, don't want to spam these issues with unrelated stuff.
